### PR TITLE
feat: thorchain lp wallet feature detection

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Brings wallet feature detection to THORChain LP.

What this PR does:

- Ensure the standalone "Add Liquidity" component narrows down the available asym types to what the wallet supports
- Ensure switching pools use a default that the wallet supports if applicable instead of the default asym pool, which may or may not work depending on wallet support

What this PR does *not*:

- Add fancy UX to disable inputs/button when the wallet doesn't have support for any a/sym type (e.g Ledger with only EVM accounts added, but DOGE/RUNE pool). To be discussed with @reallybeard and 🛢️'d in a follow-up

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- relates to  https://github.com/shapeshift/web/issues/5960

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Within the "Add Liquidity" component, only options that are supported by the wallet are enabled
- When switching assets, the enabled options are only the ones supported by the wallet, and the automagically selected one is an option supported by the wallet

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

### MM - no snaps installed

image

### Native

image

### Ledger - only RUNE added 

 
Screenshot 2024-02-07 at 19 37 37

### Ledger - only EVM accounts added 

image

### Ledger - RUNE and EVM accounts added

Screenshot 2024-02-07 at 19 42 08

### Switching between wallets with different wallet supports

https://github.com/shapeshift/web/assets/17035424/d9f31a2d-410d-455f-8cbb-128cef4ce3aa

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6148?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `develop`
  * **PR #6147** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6147?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6148** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6148?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6148?utm_source=stack-comment).

File: src/pages/ThorChainLP/components/LpType.tsx [hunk 1]
```diff
@@ -113,8 +164,15 @@ export const LpType = ({ assetId, defaultOpportunityId, onAsymSideChange }: Depo
       const radio = getRadioProps({ value: option.value })
 
       const optionAssetIds = makeAssetIdsOption(option.value as AsymSide | 'sym')
+      const walletSupportsOption = optionAssetIds.every(assetId => {
+        const isRune = assetId === thorchainAssetId
+
+        return isRune ? supportsRune : supportsAsset
+      })
+
+      const isDisabled = !walletSupportsOption
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/LpType.tsx:173]: The third guts of this

**Review by 0xApotheosis — APPROVED** [review]
We might want to narrow down the list of selectable assets, too (e.g. you shouldn't even be able to select a UTXO asset if you are on Metamask without snaps) - but for now this does precisely what it says it does 🤝

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 2]
```diff
@@ -143,6 +146,8 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
     if (opportunityId) return undefined
     if (paramOpportunityId) return paramOpportunityId
 
+    // This may be wrong if the asset doesn't support said opportunity in an asym way, but this is only used to indicate the default pool really
+    // Trying to fix this to be semantically correct will result in circular dependencies madness, don't try this if you value your sanity
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:150]: (seriously, don't)

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 3]
```diff
@@ -183,17 +188,40 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
...
+  )
+  const poolAssetAccountIds = useAppSelector(state =>
+    selectAccountIdsByAssetId(state, { assetId: poolAsset?.assetId ?? '' }),
+  )
+
+  const isSnapInstalled = useIsSnapInstalled()
+  const supportsRune =
+    walletSupportsChain({ chainId: thorchainChainId, wallet, isSnapInstalled }) &&
+    thorchainAccountIds.length > 0
+  const supportsAsset =
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: 749217c3bd29fb27174b1cfaca4b040111c620ed

**Comment by gomesalexandre** [comment]
> We might want to narrow down the list of selectable assets, too (e.g. you shouldn't even be able to select a UTXO asset if you are on Metamask without snaps) - but for now this does precisely what it says it does 🤝

Indeed that would be nice @0xApotheosis, I did discuss this with beard and give a try to having the assets *disabled* (but still displayed, as IMO we want to show the user the list of available pools, even if they can't ape into it) however given the madness of the asset selection common component, this was quite hard (read: barely impossible) to achieve unfortunately!
We'll probably want to add better support there for disabling assets, which could be reused across the app.

I'll take another look as a follow-up!

**Pull request merged**